mephisto/abstractions/providers/prolific/wrap_crowd_source.js

```diff
@@ -0,0 +1,119 @@
+}
+
+function getAssignmentId() {
+  // Prolific assignment id is passed via url params
+  urlParams = new URLSearchParams(window.location.search);
+  return urlParams.get("assignment_id");
+}
+
+function getAgentRegistration() {
+  // Prolific agents are created using the Mephisto worker_id
+  // and the supplied assignment id
+  return {
+    worker_name: getWorkerName(),
+    agent_registration_id: getAssignmentId() + "-" + getWorkerName(),
+    assignment_id: getAssignmentId(),
+    provider_type: "prolific",
+  };
+}
+
+function handleSubmitToProvider(task_data) {
+  // Prolific agents won't ever submit to a real provider
+  alert("The task has been submitted! Data: " + JSON.stringify(task_data));
+  return true;
+}
+
+const events = eventEmitter();
+
+window._MEPHISTO_CONFIG_ = window._MEPHISTO_CONFIG_ || {};
+window._MEPHISTO_CONFIG_.EVENT_EMITTER = events;
+
+window._MEPHISTO_CONFIG_.get = (property) => {
+  if (!(property in window._MEPHISTO_CONFIG_))
+    throw new Error(`${property} does not exist in window.MEPHISTO_CONFIG`);
+  else return window._MEPHISTO_CONFIG_[property];
+};
+
+window._MEPHISTO_CONFIG_.set = (property, value) => {
+  window._MEPHISTO_CONFIG_[property] = value;
+  events.emit(property, value);
+};
+
+/* === UI error handling code ======= */
+window._MEPHISTO_CONFIG_.AUTO_SUBMIT_ERRORS = false;
+window._MEPHISTO_CONFIG_.ADD_ERROR_HANDLING = false;
+window._MEPHISTO_CONFIG_.ERROR_REPORT_TO_EMAIL = null;
+
+let numErrorsCaught = 0;
+let numErrorsReported = 0;
+let userDisabledErrorPrompts = false;
+// Adding event listener instead of using window.onerror prevents the error to be caught twice
+window.addEventListener("error", function (event) {
+  if (event.error.hasBeenCaught === true) {
+    return false;
+  }
+  event.error.hasBeenCaught = true;
+  numErrorsCaught += 1;
+  if (
+    window._MEPHISTO_CONFIG_.ADD_ERROR_HANDLING &&
+    !userDisabledErrorPrompts
+  ) {
+    const fullErrorMessage = `${event.error.message}\n\n${event.error.stack}`;
+    if (
+      confirm(
+        "An error has been detected and may affect the function of this task. Would you like to report the error?\n\nDetails: " +
+          event.error.message
+      )
+    ) {
+      numErrorsReported += 1;
+      if (window._MEPHISTO_CONFIG_.ERROR_REPORT_TO_EMAIL) {
+        prompt(
+          `Please copy & paste the following in an email to: ${window._MEPHISTO_CONFIG_.ERROR_REPORT_TO_EMAIL}`,
+          fullErrorMessage
+        );
+      } else {
+        prompt(
+          `Please copy & paste the following in an email to the task owner:`,
+          fullErrorMessage
+        );
+      }
+    }
+
+    if (numErrorsCaught > 1) {
+      userDisabledErrorPrompts = confirm(
+        "We noticed you have gotten more than one of these error alerts. Would you like to hide these alerts going forward?"
+      );
+      console.log(userDisabledErrorPrompts);
+    }
+  } else {
+    // should we do auto-submission of errors?
+  }
+});
```

Comment:
@pringshia is there a target for moving these (and the `event_emitter`) into the base `mephisto_task` library? I don't imagine these change provider-to-provider, and they're bloating these files.

Reply:
We needed the even emitter to enable functionality for all tasks including non-react static tasks #692, and so the emitter was introduced via #879. I don't know how else we'd add it. As is now,  `mephisto-task` is mostly a React-based package for React-based tasks.